### PR TITLE
Env Var Feedback

### PR DESCRIPTION
**Pull request by lovincyrus**
Address https://rilldata.slack.com/archives/C07NUPQ02EA/p1733432093128959

- [x] https://www.notion.so/UXQA-Environment-Variables-154ba33c8f57809fb02fdd7d7a834c28?d=154ba33c8f5780a38f91001cb18df19e&pvs=4#154ba33c8f5780f3a754c60b3ed0888a
- [x] feedback
- [x] more testing

### CODE REVIEW

**Review by ericpgreen2 — DISMISSED** [review]
UXQA:
- [x] I get a validation warning "You must select at least one environment" when I open & close the "Edit variable" form, then open the form a second time: [Jam](https://jam.dev/c/57f43139-e674-48b8-a84e-cbbfd9d7c54e)
- [x] When I edit the name of an environment variable so that it would conflict with another,  I get a validation error (expected) and the name flips back to its original name (unexpected): [Jam](https://jam.dev/c/6ca9fe04-1415-4d97-8e04-63a54e841ce9). It's confusing that the validation error is presented for the the original valid name, not for the edited invalid name.

File: web-admin/src/features/projects/environment-variables/AddDialog.svelte [hunk 1]
```diff
@@ -190,43 +190,119 @@
...
+  // QUOTE="String with \"quotes\"" -> { key: "QUOTE", value: 'String with "quotes"' }
+  //
+  // Special cases:
+  // EMPTY=""                      -> { key: "EMPTY", value: "" }
+  // EMPTY_SINGLE=''               -> { key: "EMPTY_SINGLE", value: "" }
+  // EQUALS="key=value"            -> { key: "EQUALS", value: "key=value" }
+  // MULTILINE="first\nsecond"     -> { key: "MULTILINE", value: "first\nsecond" }
+  // SPACES="  trim me  "          -> { key: "SPACES", value: "  trim me  " }
+  // UNICODE="🚀"                  -> { key: "UNICODE", value: "🚀" }
+  function parseFile(contents: string) {
```
[the remark this answers is not in the document]
Comment (lovincyrus) [reply web-admin/src/features/projects/environment-variables/AddDialog.svelte:209]: good call

**Comment by ericokuma** [comment]
Adding that the validation errors (for key name duplication) doesn't reset when I enter a new name and therefore am not able to Edit:

https://www.loom.com/share/448bbb48880347a3b81d0847f2ee5ab4?sid=5630da38-0552-496a-bc2d-848c89eef872

**Comment by lovincyrus** [comment]
Addressed all the recent feedback! @ericokuma @ericpgreen2

**Comment by ericpgreen2** [comment]
UXQA: I'm unable to add a key for a second environment: [Jam](https://jam.dev/c/87797b42-f822-43b6-8ea5-d9a78b1d3c07)

**Comment by ericokuma** [comment]
Ah yeah, good find. That is a bit of an odd behavior. 

Ideal:
- Checking the key name for duplicates should occur either after a user clicks out of the key name input field or selecting/deselecting environments
- This should be coupled with the disabling/enabling of the Create/Edit buttons

Acceptable solution:
- Always enable the Create/Edit button and do the validation check only upon user clicking on that button i.e. do not display validation error until button has been clicked

**Comment by ericokuma** [comment]
A couple of issues I found:
https://www.loom.com/share/a2c39bf392114c9ba3f5ada4150b92ae?sid=cd393837-5fcf-4c4b-be0d-187da4b53530

1. When adding a new environment and inputting a key name that matches an existing key (with the exception of capitalizing one or more letters so that it's different in terms of case), the form doesn't detect and block creation

2. Instead, it edits the value of the existing key with the new value inputting with the duplicative entry

**Comment by lovincyrus** [comment]
> A couple of issues I found: https://www.loom.com/share/a2c39bf392114c9ba3f5ada4150b92ae?sid=cd393837-5fcf-4c4b-be0d-187da4b53530
> 
> 1. When adding a new environment and inputting a key name that matches an existing key (with the exception of capitalizing one or more letters so that it's different in terms of case), the form doesn't detect and block creation
> 2. [BLOCKER] Instead, it edits the value of the existing variable with the new value inputting with the duplicative entry (looks like it also changes the environments selected as well for the existing variable)

I looked up the API Schema. It appears that variable name is insensitive.

https://github.com/rilldata/rill/pull/5929/files#diff-5135893de718e4a71cefcbda1d7cece93dbaad8859c7082841e26ccadbee0136R2093

`ON CONFLICT (project_id, environment, lower(name)) DO UPDATE SET`

https://github.com/rilldata/rill/blob/main/proto/gen/rill/admin/v1/admin.swagger.yaml#L4755
```
  name:
    type: string
    description: Variable name (case insensitive).
```

Case insensitivity is already handled at the DB level. When we call `UpdateProjectVariables` when adding new variables, "john", "John", "JOHN". They are all treated as the same variable. @ericokuma

**Pull request merged**